src/components/InputNumber.tsx

```diff
@@ -60,25 +67,25 @@ export default class InputNumber extends React.Component {
     }
 
     this.setState({
-      dirtyValue: newValue === "" ? undefined : newValue,
+      dirtyValue: newValue === "" ? undefined : value,
```

Comment:
these changes between value/newValue appear like it could change some logic - is it a result of automatic conversion?

Reply:
More of a typing problem. Value is types correctly and is initialized with newValue in the beginning of the method.
This was not automatic.